pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -791,15 +780,48 @@ Set<Integer> getPartitionIds(StreamConfig streamConfig)
     }
   }
 
+  @VisibleForTesting
+  Set<Integer> getPartitionIds(List<StreamConfig> streamConfigs, IdealState idealState) {
+    Set<Integer> partitionIds = new HashSet<>();
+    boolean allPartitionIdsFetched = true;
+    for (int i = 0; i < streamConfigs.size(); i++) {
+      final int index = i;
+      try {
+        partitionIds.addAll(getPartitionIds(streamConfigs.get(index)).stream()
+            .map(partitionId -> IngestionConfigUtils.getPinotPartitionIdFromStreamPartitionId(partitionId, index))
+            .collect(Collectors.toSet()));
+      } catch (Exception e) {
+        allPartitionIdsFetched = false;
+        LOGGER.warn("Failed to fetch partition ids for stream: {}", streamConfigs.get(i).getTopicName(), e);
+      }
+    }
+
+    // Some fetches failed, so ensure we do not miss any partition ids
```

Comment:
Please explain in high level how to deal with partial fetch failures. From the code, it looks like we reuse the existing partition and then fetch only new partitions. But it is better to explain clearly what is done here.